clc/modules/object-storage-common/src/main/java/com/eucalyptus/objectstorage/util/OSGUtil.java

```diff
@@ -126,16 +137,21 @@ public static BaseMessage convertErrorMessage(EucalyptusErrorMessageType errorMe
 
   private static BaseMessage convertException(String correlationId, Throwable ex) {
     BaseMessage errMsg;
-    if (ex instanceof ObjectStorageException) {
+    if (ex instanceof S3ExtendedException){
+      S3ExtendedException e = (S3ExtendedException) ex;
+      errMsg =
+          new ObjectStorageErrorMessageExtendedType(e.getMessage(), e.getCode(), e.getStatus(), e.getResourceType(), e.getResource(), correlationId,
+              Internets.localHostAddress(), e.getLogData(), e.getRequestMethod());
+    } else if (ex instanceof ObjectStorageException) {
```

Comment:
Refer to the comment about adding requestMethod to S3Exception class and creating a new constructor - one that takes in requestMethod as an argument in addition to others